### PR TITLE
Allow any wall shape for Kiva Foundations

### PR DESCRIPTION
**Pull request by nealkruis**
Pull request overview
---------------------
Removes the restriction of only quadrilateral walls for Kiva Foundations.

Work completed under the Big Ladder Contribution License Agreement.

### Review Checklist
This will not be exhaustively relevant to every PR.
 - [ ] Code style (parentheses padding, variable names)
 - [ ] Functional code review (it has to work!)
 - [ ] CI status: all green or justified
 - [ ] Performance: CI Linux results include performance check -- verify this
 - [ ] Unit Test(s)
 - C++ checks:
   - [ ] Argument types
 - [ ] Documentation changes in place
 - [ ] Changed docs build successfully

### CODE REVIEW

**Comment by nealkruis** [comment]
@Myoldmopar let me know if someone else can/should review. I'm also seeing all integration tests failing on Mac, but it doesn't seem to be related to this branch. Any idea what's going on there?

File: doc/input-output-reference/src/overview/group-advanced-surface-concepts.tex [hunk 1]
```diff
@@ -523,7 +523,8 @@ \subsection{Foundation:Kiva}\label{foundationkiva}
   However, multiple floors may exist in the same thermal zone so long as
   they reference separate Foundation:Kiva objects.
 \item
-  All foundation wall surfaces must be quadrilateral.
+  Foundation wall surfaces that are not triangular or quadrilateral may translate
+  well into the two-dimensional finite difference context.
```
Comment (mjwitte) [inline doc/input-output-reference/src/overview/group-advanced-surface-concepts.tex]: Did you mean "may *not* translate well . . ."?

Comment (nealkruis) [reply doc/input-output-reference/src/overview/group-advanced-surface-concepts.tex]: Yes. I did.

**Comment by nealkruis** [comment]
@Myoldmopar this is coming back good except some seemingly unrelated regression test failures. Do you have time to review this?

**Comment by nealkruis** [comment]
@mjwitte @Myoldmopar Any thoughts on this?

**Comment by mjwitte** [comment]
@nealkruis Lots of PRs in the queue. Priority at this point is to review PRs with IDD changes.

**Comment by nrel-bot-3** [comment]
@nealkruis @lgentile it has been 14 days since this pull request was last updated.

**Comment by Myoldmopar** [comment]
I don't have any issue with the code, it looks like it had some diffs that were just due to being behind develop?  I'll pull develop in here as I think merges are pretty much done for the day and we'll see if we can drop this in tonight or tomorrow morning.

**Pull request merged**